graphqlapi/src/main/java/io/stargate/graphql/web/resources/FilesResource.java

```diff
@@ -98,33 +101,51 @@ public Response getSchema(
       throws Exception {
 
     if (!DmlResource.KEYSPACE_NAME_PATTERN.matcher(keyspace).matches()) {
-      LOG.warn("Invalid keyspace in URI, this could be an XSS attack: {}", keyspace);
+      LOG.warn("Malformed keyspace in URI, this could be an XSS attack: {}", keyspace);
       // Do not reflect back the value
-      return Response.status(Response.Status.BAD_REQUEST).entity("Invalid keyspace name").build();
-    }
-
-    if (!isAuthorized(token, httpRequest)) {
-      return Response.status(Response.Status.UNAUTHORIZED).build();
+      return Response.status(Response.Status.BAD_REQUEST).entity("Malformed keyspace name").build();
     }
 
     UUID versionUuid = null;
     if (version != null) {
       try {
         versionUuid = UUID.fromString(version);
       } catch (IllegalArgumentException e) {
-        return notFound(keyspace, version);
+        LOG.warn("Malformed version in URI, this could be an XSS attack: {}", version);
+        return Response.status(Response.Status.BAD_REQUEST).entity("Malformed version").build();
       }
     }
 
-    SchemaSource schemaSource =
-        schemaSourceDao.getSingleVersion(keyspace, Optional.ofNullable(versionUuid));
-    if (schemaSource == null) {
-      return notFound(keyspace, version);
-    }
+    try {
+      Map<String, String> headers = RequestToHeadersMapper.getAllHeaders(httpRequest);
+      AuthenticationSubject authenticationSubject =
+          authenticationService.validateToken(token, headers);
+      authorizationService.authorizeSchemaRead(
+          authenticationSubject,
+          Collections.singletonList(SchemaSourceDao.KEYSPACE_NAME),
+          Collections.singletonList(SchemaSourceDao.TABLE_NAME),
+          SourceAPI.GRAPHQL,
+          ResourceKind.TABLE);
 
-    return Response.ok(schemaSource.getContents())
-        .header("Content-Disposition", "inline; filename=" + createFileName(schemaSource))
-        .build();
+      DataStore dataStore =
+          dataStoreFactory.create(
+              authenticationSubject.asUser(),
+              DataStoreOptions.builder().putAllCustomProperties(headers).build());
+      SchemaSource schemaSource =
+          new SchemaSourceDao(dataStore)
+              .getSingleVersion(keyspace, Optional.ofNullable(versionUuid));
```

Comment:
This is the main change: create a request-scoped datastore every time, instead of using a global, non-scoped one.